src/provider.coffee

```diff
@@ -5,6 +5,25 @@ extensions        = require './extensions'
 
 
 
+paramMatches = (param, duplicate) ->
+  if Array.isArray param
```

Comment:
We're not trying to match the array though are we?

_http://example.com?foo=a&foo=b&foo=c&bar=q_

_foo_ is parsed into an array, but the duplication is against _(foo, a)_, _(foo, b)_, _(foo, c)_, not _(foo, [a,b, c])_

That said, if they're guaranteed to pass every query param as part of the body as well - maybe we should match the whole array? Or maybe we can just throw out the query params?